lang/java/tools/src/main/java/org/apache/avro/tool/SpecificCompilerTool.java

```diff
@@ -62,23 +62,23 @@ public int run(InputStream in, PrintStream out, PrintStream err, List<String> ar
 
     int arg = 0;
 
-    if ("-encoding".equals(args.get(arg))) {
+    if (args.contains("-encoding")) {
```

Comment:
I think the checks for "-encoding" and "-templateDir" are incorrect -- the index `arg` won't be positioned correctly to assign the value.

Reply:
Hi @RyanSkraba you are correct about that and I have made changes accordingly so please do review and test it when you have time.